src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -256,4 +256,36 @@ void convertAVFrameToDecodedOutputOnCuda(
           << " took: " << duration.count() << "us" << std::endl;
 }
 
+// inspired by https://github.com/FFmpeg/FFmpeg/commit/ad67ea9
```

Comment:
Let's provide some more context for why we're doing this. We definitely want to reference this FFmpeg GitHub issue, but let's also say that something similar to your summary: we have to do this because of an FFmpeg bug where hardware decoding is not appropriately set, so we just go off and find the matching codec for the CUDA device.